designs/exportable-subflow/README.md

```diff
@@ -24,28 +24,52 @@ are in the roadmap. (*TODO: add a link to the design doc once added*).
 
 ### Details
 
-#### Exporting Subflow as Node
-
-Node can be exported as JSON code or converted to a Node.
+Node can be exported as JSON code (*SUBFLOW Module*) or converted to a Node (*SUBFLOW Node*). In this document, design for *SUBFLOW Module* is described.
 
+#### Exporting SUBFLOW Module
 
 ![Subflow-export-UI](Subflow-export-UI.png)
 
-By clicking **Export** button on Subflow Setting Panel of Subflow template, Subflow Export Panel is shown.
-
-In this panel, new node information such as node name, version, etc. can be specified. There is **As JSON** check box on this panel.  If this is checked, JSON export panel is shown.  Otherwise, new node is installed on  editor palette.  
+By clicking **Export SUBFLOW** button on SUBFLOW template setting panel, SUBFLOW export panel is shown.
 
-New node created from SUBFLOW is sealed.  Thus, its internal flow or UI definition can not be edited.
+In this panel, new node information such as node name, version, etc. can be specified. There is **Include All SUBFLOW** check box on this panel.  If this is checked, SUBFLOWs used within the exported SUBFLOW are recursively included in exported SUBFLOW Module.  If **Editable** check box on this panel is checked, SUBFLOW template can be edited on imported Node-RED editor.  Otherwise, the imported SUBLOW template is sealed. Thus, its internal flow or UI definition can not be edited.
 
-#### Importing SUBFLOW Node
+#### Importing SUBFLOW Module
 
-Node represented as JSON data can be imported in Node-RED environment from palette management UI.
+Node represented as JSON data can be imported in Node-RED environment from import new node panel.
 
 ![Subflow-import-UI](Subflow-import-UI.png)
 
-There is **Install(JSON)** button on top of the panel.  After importing JSON, new node is installed on editor palette.
+#### Uninstalling SUBFLOW Module
+
+SUBFLOW Module are listed in *Manage Palette > Palette > Nodes* tab.  Similar to normal nodes, *SUBFLOW Module*s listed in this tab will have **remove** button.  If a SUBFLOW Module includes other SUBFLOW Modules and is sealed, contained SUBFLOW Modules are also deleted.
+
+#### SUBFLOW Module Format
+
+Exported SUBFLOW Module is represented by JSON array format which is similar to current flow representation.  
+
+Following is a list of properties exported SUBFLOW node in SUBFLOW Module may contain (+ means that property is optional):
+
+|      | name    | type   | description                        |
+| ---- | ------- | ------ | ---------------------------------- |
+| 1    | flow    | array  | array of contained nodes in SUBFLOW |
+| 2    | sealed+ | bool   | SUBFLOW template is not editable   |
+| 3    | hidden+ | bool   | SUBFLOW template is hidden         |
```

Comment:
What is the `hidden` property for? I can't remember discussing that (apologies if we did) but this design doesn't explain what it means.

Reply:
Add description of `hidden` property to design note.
It is used for SUBFLOW contained in sealed SUBFLOW.